tests/e2e/specs/in-game/gameOver.spec.js

```diff
@@ -388,3 +392,233 @@ describe('Stalemates', () => {
+    });
+
+    // 3rd game: Ends via requested stalemate
+    setupGameAsP0(true, true);
+    // Request stalemate
+    cy.get('#game-menu-activator').click();
+    cy.get('#game-menu').should('be.visible').get('[data-cy=stalemate-initiate]').click();
+    cy.get('#request-gameover-dialog')
+      .should('be.visible')
+      .get('[data-cy=request-gameover-confirm]')
+      .click();
+    cy.get('#waiting-for-opponent-stalemate-scrim').should('be.visible');
+    // Opponent confirms
+    cy.stalemateOpponent();
+    cy.get('[data-cy=gameover-go-home]').click();
+    cy.url().should('not.include', '/game');
+
+    // Validate match data
+    cy.request('http://localhost:1337/match').then((res) => {
+      expect(res.body.length).to.eq(2);
+      const [, currentMatch] = res.body;
+      expect(currentMatch.player1.id).to.eq(this.playerOneId);
+      expect(currentMatch.player2.id).to.eq(this.playerTwoId);
+      expect(currentMatch.startTime).to.be.greaterThan(0);
+      expect(currentMatch.games.length).to.eq(3);
+      // Match is incomplete
+      expect(currentMatch.endTime).to.eq(0);
+      expect(currentMatch.winner).to.eq(null);
+      expect(currentMatch.games[2].result).to.eq(2);
+      cy.log('Match data is correct after third game', res);
+    });
+
+    // 4th game: stalemate due to passing
+    setupGameAsP0(true, true);
+    cy.loadGameFixture({
+      p0Hand: [Card.SEVEN_OF_CLUBS],
+      p0Points: [Card.SEVEN_OF_DIAMONDS, Card.SEVEN_OF_HEARTS],
+      p0FaceCards: [],
+      p1Hand: [],
+      p1Points: [],
+      p1FaceCards: [],
+    });
+    cy.get('[data-player-hand-card]').should('have.length', 1);
+    cy.log('Fixture loaded');
+
+    cy.deleteDeck();
+    cy.log('Drawing last two cards');
+    cy.get('#deck').should('contain', '(2)').click();
+    cy.drawCardOpponent();
+    cy.log('Deck empty');
+    cy.get('#deck').should('contain', '(0)').should('contain', 'PASS').click();
+    cy.get('#turn-indicator').contains("OPPONENT'S TURN");
+    cy.passOpponent();
+    cy.get('#turn-indicator').contains('YOUR TURN');
+    cy.get('#deck').should('contain', '(0)').should('contain', 'PASS').click();
+    cy.get('[data-cy=gameover-go-home]').click();
+    cy.url().should('not.include', '/game');
+
+    // Validate match data
+    cy.request('http://localhost:1337/match').then((res) => {
+      expect(res.body.length).to.eq(2);
+      const [, currentMatch] = res.body;
+      expect(currentMatch.player1.id).to.eq(this.playerOneId);
+      expect(currentMatch.player2.id).to.eq(this.playerTwoId);
+      expect(currentMatch.startTime).to.be.greaterThan(0);
+      expect(currentMatch.games.length).to.eq(4);
+      expect(currentMatch.games[3].result).to.eq(2);
+      // Match is incomplete
+      expect(currentMatch.endTime).to.eq(0);
+      expect(currentMatch.winner).to.eq(null);
+      cy.log('Match data is correct after fourth game', res);
+    });
+
+    // 5th Game: player wins via points and wins match
```

Comment:
Todo: play an unranked game and confirm it doesn't get associated with the match